core/src/main/java/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -306,6 +310,16 @@ private Answer execute(DiagnosticsCommand cmd) {
         return new DiagnosticsAnswer(cmd, result.isSuccess(), result.getDetails());
     }
 
+    private Answer execute(PrepareFilesCommand cmd) {
+        String fileList = String.join(" ", cmd.getFilesToRetrieveList());
+        _eachTimeout = Duration.standardSeconds(cmd.getTimeout());
+        final ExecutionResult result = _vrDeployer.executeInVR(cmd.getRouterAccessIp(), VRScripts.RETRIEVE_DIAGNOSTICS, fileList, _eachTimeout);
+        if (result.isSuccess()){
+            return new PrepareFilesAnswer(cmd, true, result.getDetails());
+        }
+        return new PrepareFilesAnswer(cmd, false, result.getDetails());
```

Comment:
or on this failure case?

Reply:
Same as above, VR script execution outcome is propagated back to the service layer